### PR TITLE
Fix the calculation of rates in ConnectionStatistics

### LINKED ISSUE #5783: Fix ConnectionStatistics.*Rate() methods
**Jetty version**
9.4.x

**Description**
`ConnectionStatistics.*Rate()` methods do a `getAndSet()` on the timestamp that is obviously wrong (what was I thinking?!?).

These methods should just do a `get()` of the startup timestamp and use it to calculate the elapsed time, and then do the rate calculations.
Having a single startup timestamp will also get rid of the useless 4 timestamp fields we have now.

### PR DESCRIPTION
**Pull request by lachlan-roberts**
closes #5783

The calculation for the rate was previously done as `getTotal()  / elapsedTime`.
When it should really be `getIncreaseOverElapsedTime() / elapsedTime`.

### CODE REVIEW

**Comment by lachlan-roberts** [comment]
@sbordet another problem with this is that `ConnectionStatistics` doesn't collect live statistics from connections. It only updates when the connection closes, so its probably not going to give you great statistics for rates in websocket. You'll probably be using long lasting websocket connections so the rate values would probably be pretty useless.

**Comment by gregw** [comment]
Maybe have some kind of 'takeStats' method that updates all the totals and remembers the levels in each connection so it doesn't get double counted.
call `takeStats` only if somebody queries the stats... not sure if all the wiring necessary is there... but look

File: jetty-util/src/main/java/org/eclipse/jetty/util/statistic/RateCounter.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.LongAdder;
+
+/**
+ * Gives the same basic functionality of {@link LongAdder} but allows you to check
+ * the rate of increase of the sum since the last call to {@link #getRate()};
+ */
+public class RateCounter
+{
+    private final LongAdder _total = new LongAdder();
```
Comment (gregw) [inline jetty-util/src/main/java/org/eclipse/jetty/util/statistic/RateCounter.java:30]: Is the total really necessary in this class?
It is entirely independent so no benefit being in here.  If you just want a rate you don't need it, if you also want the total then just have a LongAdder in addition to a RateCounter.

**Review by gregw — APPROVED** [review]
I'm +0 as I don't really like the total being included in the RateCounter.
I can see how it makes the ConnectionStatistics a little neater, but it is kind of unrelated.

File: jetty-util/src/main/java/org/eclipse/jetty/util/statistic/RateCounter.java (hunk 1 above)
Comment (lachlan-roberts) [reply jetty-util/src/main/java/org/eclipse/jetty/util/statistic/RateCounter.java:30]: I removed the total from the RateCounter. 

You now have to explicitly call  `RateCounter.reset()` to reset it, it doesn't do this automatically on the `RateCounter.getRate()` call anymore, the reset is called from the `ConnectionStatistics.getRate()` methods.

**Pull request merged**